cpp/CMakePresets.json

```diff
@@ -527,16 +554,45 @@
       "displayName": "Benchmarking build with everything enabled",
       "cacheVariables": {}
     },
+    {
+      "name": "ninja-debug-asan",
+      "inherits": [
+        "ninja-debug",
+        "sanitizer-asan"
+      ],
+      "displayName": "Debug ASAN build with tests and more optional components",
+      "cacheVariables": {}
+    },
+    {
+      "name": "ninja-debug-tsan",
+      "inherits": [
+        "ninja-debug",
+        "sanitizer-tsan"
+      ],
+      "displayName": "Debug TSAN build with tests and more optional components",
+      "cacheVariables": {}
+    },
+    {
+      "name": "ninja-debug-ubsan",
+      "inherits": [
+        "ninja-debug",
+        "sanitizer-ubsan"
+      ],
+      "displayName": "Debug UBSAN build with tests and more optional components",
+      "cacheVariables": {}
+    },
     {
       "name": "fuzzing",
-      "inherits": "base",
+      "inherits": [
+        "base",
+        "sanitizer-asan",
```

Comment:
I think this might improve the chance to reveal issues that are hidden by jemalloc/mimalloc being enabled before.